File: Android/tuilivekit/livekit/src/main/java/com/trtc/uikit/livekit/features/livelist/access/SingleColumnListViewAdapter.java
package com.trtc.uikit.livekit.features.livelist.access;

import android.content.Context;
import android.view.View;

import com.tencent.cloud.tuikit.engine.extension.TUILiveListManager;
import com.trtc.uikit.livekit.features.livelist.LiveListViewDefine;

public class SingleColumnListViewAdapter implements LiveListViewDefine.LiveListViewAdapter {
    private final Context mContext;

    public SingleColumnListViewAdapter(Context context) {
        mContext = context;
    }

    @Override
    public View createLiveInfoView(TUILiveListManager.LiveInfo liveInfo) {
        SingleColumnWidgetView widgetView = new SingleColumnWidgetView(mContext);
        widgetView.init(liveInfo);
        return widgetView;
    }

    @Override
    public void updateLiveInfoView(View view, TUILiveListManager.LiveInfo liveInfo) {
        SingleColumnWidgetView widgetView = (SingleColumnWidgetView) view;
        widgetView.updateLiveInfoView(liveInfo);
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/features/livelist/access/SingleColumnListViewAdapter.kt
package com.trtc.uikit.livekit.features.livelist.access

import android.content.Context
import android.view.View
import com.tencent.cloud.tuikit.engine.extension.TUILiveListManager
import com.trtc.uikit.livekit.features.livelist.LiveListViewAdapter

class SingleColumnListViewAdapter(private val context: Context) : LiveListViewAdapter {

    override fun createLiveInfoView(liveInfo: TUILiveListManager.LiveInfo): View {
        val widgetView = SingleColumnWidgetView(context)
        widgetView.init(liveInfo)
        return widgetView
    }

    override fun updateLiveInfoView(view: View, liveInfo: TUILiveListManager.LiveInfo) {
        val widgetView = view as SingleColumnWidgetView
        widgetView.updateLiveInfoView(liveInfo)
    }
}
